src/trackers.js

```diff
@@ -1,227 +1,254 @@
-const abp = require('abp-filter-parser')
 const utils = require('./utils')
-const trackersWithParentCompany = require('../data/generated/trackers-with-parent-company')
-const entityMap = require('../data/generated/entity-map')
-const surrogates = require('./surrogates')
-
-const blockSettings = ['Advertising', 'Analytics']
+const tldjs = require('tldjs')
+const btoa = require('btoa')
 
 class Trackers {
     addLists (lists) {
-        this.entityList = lists.entityList
-        this.whitelist = {}
+        this.entityList = this.processEntityList(lists.entityList)
+        this.trackerList = this.processTrackerList(lists.trackerList)
+        this.surrogateList = this.processSurrogateList(lists.surrogates)
+        this.exceptionTypes = ['domains', 'types']
+    }
 
-        abp.parse(lists.whitelist, this.whitelist)
+    processTrackerList (data) {
+        Object.keys(data).forEach(name => {
+            if (data[name].rules) {
+                for (let i in data[name].rules) {
+                    // fix this later. make a toString to serialize RegExp?
+                    data[name].rules[i].ruleStr = data[name].rules[i].rule
+                    data[name].rules[i].rule = new RegExp(data[name].rules[i].rule, 'ig')
+                }
+            }
+        })
+        return data
     }
 
-    isTracker (urlToCheck, currLocation, requestType, ops) {
-        ops = ops || {}
+    processEntityList (data) {
+        let processed = {}
+        Object.keys(data).forEach(entity => {
+            data[entity].properties.forEach(domain => {
+                processed[domain] = entity
+            })
+        })
+        return processed
+    }
 
-        if (!this.entityList || !this.whitelist) {
-            throw new Error('tried to detect trackers before rules were loaded')
-        }
+    processSurrogateList (text) {
+        const b64dataheader = 'data:application/javascript;base64,'
+        let surrogateList = {}
+        let splitSurrogateList = text.trim().split('\n\n')
+
+        splitSurrogateList.forEach(sur => {
+            // remove comment lines
+            let lines = sur.split('\n').filter((line) => {
+                return !(/^#.*/).test(line)
+            })
 
-        let currLocationDomain = utils.getDomain(currLocation)
-        let hostnameToCheck = utils.extractHostFromURL(urlToCheck)
-        let parsedUrl = { domain: utils.getDomain(urlToCheck) }
+            // remove first line, store it
+            let firstLine = lines.shift()
 
-        if (!hostnameToCheck) {
-            return false
-        }
+            // take identifier from first line
+            let pattern = firstLine.split(' ')[0]
+            let b64surrogate = btoa(lines.join('\n'))
+            surrogateList[pattern] = b64dataheader + b64surrogate
+        })
+        return surrogateList
+    }
 
-        let urlSplit = hostnameToCheck.split('.')
+    getTrackerData (urlToCheck, siteUrl, request, ops) {
+        ops = ops || {}
```

Comment:
Not used but this is where we would pass in settings in the real implementation.